packaging/greenboot/microshift-running-check.sh

```diff
@@ -0,0 +1,239 @@
+    local is_failed=$(systemctl is-failed microshift.service)
+    local is_active=$(systemctl is-active microshift.service)
+
+    # Terminate the script in case of a failed service - nothing to wait for
+    if [ "${is_failed}" = "failed" ] ; then
+        echo "The microshift.service systemd unit is failed. Terminating..."
+        kill -TERM ${SCRIPT_PID}
+    fi
+    # Check the service activity
+    [ "${is_active}" = "active" ] && return 0
+    return 1
+}
+
+# Check if MicroShift API 'readyz' and 'livez' health endpoints are OK
+# 
+# args: None
+# return: 0 if all API health endpoints are OK, or 1 otherwise
+function microshift_health_endpoints_ok() {
+    local check_rd=$(${OCGET_CMD} --raw='/readyz?verbose' | awk '$2 != "ok"')
+    local check_lv=$(${OCGET_CMD} --raw='/livez?verbose'  | awk '$2 != "ok"')
+
+    [ "${check_rd}" != "readyz check passed" ] && return 1
+    [ "${check_lv}" != "livez check passed"  ] && return 1
+    return 0
+}
+
+# Check if any MicroShift pods are in the 'Running' status
+# 
+# args: None
+# return: 0 if any pods are in the 'Running' status, or 1 otherwise
+function any_pods_running() {
+    local count=$(${OCGET_CMD} pods ${OCGET_OPT} -A 2>/dev/null | awk '$4~/Running/' | wc -l)
+
+    [ "${count}" -gt 0 ] && return 0
+    return 1
+}
+
+# Check if all the MicroShift pod images in a given namespace are downloaded.
+#
+# args: None
+# env1: 'CHECK_PODS_NS' environment variable for the namespace to check
+# return: 0 if all the images in a given namespace are downloaded, or 1 otherwise
+function namespace_images_downloaded() {
+    local ns=${CHECK_PODS_NS}
+
+    local images=$(${OCGET_CMD} pods ${OCGET_OPT} -n ${ns} -o jsonpath="{.items[*].spec.containers[*].image}" 2>/dev/null)
+    for i in ${images} ; do
+        # Return an error on the first missing image
+        local cimage=$(crictl image -q ${i})
+        [ -z "${cimage}" ] && return 1
+    done
+
+    return 0
+}
+
+# Check if a given number of MicroShift pods in a given namespace are in the 'Ready' status,
+# terminating the script with the SIGTERM signal if more pods are ready than expected.
+#
+# args: None
+# env1: 'CHECK_PODS_NS' environment variable for the namespace to check
+# env2: 'CHECK_PODS_CT' environment variable for the pod count to check
+# return: 0 if the expected number of pods are ready, or 1 otherwise
+function namespace_pods_ready() {
+    local ns=${CHECK_PODS_NS}
+    local ct=${CHECK_PODS_CT}
+
+    local status=$(${OCGET_CMD} pods ${OCGET_OPT} -n ${ns} -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}' 2>/dev/null)
+    local tcount=$(echo $status | grep -o True  | wc -l)
+    local fcount=$(echo $status | grep -o False | wc -l)
+
+    # Terminate the script in case more pods are ready than expected - nothing to wait for
+    if [ "${tcount}" -gt "${ct}" ] ; then
+        echo "The number of ready pods in the '${ns}' namespace is greater than the expected '${ct}' count. Terminating..."
```

Comment:
Do we get any debug output?

Reply:
Not at this time. We could add some code in the EXIT handler to print info on FAILURE.
I would do it in a separate PR though because we also need to define what we'd like to see there.

Reply:
No prob